fuzz-handle_onion_message: variable 'daemon' is used uninitialized whenever 'if' condition is true

Turned up in fuzzing infra:
```bash
 #10 479.9 tests/fuzz/fuzz-handle_onion_message.c:78:6: error: variable 'daemon' is used uninitialized whenever 'if' condition is true [-Werror,-Wsometimes-uninitialized]
#10 479.9    78 |         if (setjmp(fuzz_env) != 0)
#10 479.9       |             ^~~~~~~~~~~~~~~~~~~~~
#10 479.9 /usr/include/setjmp.h:49:21: note: expanded from macro 'setjmp'
#10 479.9    49 | #define setjmp(env)     _setjmp (env)
#10 479.9       |                         ^
#10 479.9 tests/fuzz/fuzz-handle_onion_message.c:104:6: note: uninitialized use occurs here
#10 479.9   104 |         if (daemon)
#10 479.9       |             ^~~~~~
#10 479.9 tests/fuzz/fuzz-handle_onion_message.c:78:2: note: remove the 'if' if its condition is always false
#10 479.9    78 |         if (setjmp(fuzz_env) != 0)
#10 479.9       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~
#10 479.9    79 |                 goto cleanup;
#10 479.9       |                 ~~~~~~~~~~~~
#10 479.9 tests/fuzz/fuzz-handle_onion_message.c:81:23: note: initialize the variable 'daemon' to silence this warning
#10 479.9    81 |         struct daemon *daemon;
#10 479.9       |                              ^
#10 479.9       |                               = NULL
#10 479.9 1 error generated.
#10 479.9 make: *** [Makefile:332: tests/fuzz/fuzz-handle_onion_message.o] Error 1
```
cc @dergoegge

--- a/tests/fuzz/fuzz-handle_onion_message.c
+++ b/tests/fuzz/fuzz-handle_onion_message.c
@@ -75,12 +75,12 @@ void init(int *argc, char ***argv)
 
 void run(const uint8_t *data, size_t size)
 {
-	if (setjmp(fuzz_env) != 0)
-		goto cleanup;
-
-	struct daemon *daemon;
+	struct daemon *daemon = NULL;
 	struct peer *peer;
 	struct pubkey dummy_key;
+
+	if (setjmp(fuzz_env) != 0)
+		goto cleanup;
 
 	memset(&dummy_key, 'c', sizeof(dummy_key));
 

